### PR TITLE
Fix unexpected operation of AirloopHVAC:UnitarySystem fan when only one coil is present

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
When the AirloopHVAC:UnitarySystem has only 1 coil, the fan operates at max air flow rate when the missing coil would have operated. This causes excess fan power for single mode operation. This corrects issue #6200.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)
 - C++ checks:
   - [x] Argument types
   - [x] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes: n/a
 n/a If new idf included, locally check the err file and other outputs
 - [x] Documentation changes in place
 - [x] Changed docs build successfully
 n/a ExpandObjects changes?
 n/a If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
One file failed, and not surprisingly, UnitarySystem_DXCoilSystemAuto. this is the old HVACDXSystem where 2 systems are on the main branch with a dangling fan. This does appear to work as expected. Using the set point based control in this file the VAV terminal units are supposed to set the air flow. But this file does not use VAV, it uses the uncontrolled TU. So air flow will be constant in both cooling and heating based on the sum of the TU flow rates. This is what is happening here. What could be better is if the system operated at the actual heating and cooling air flow rates but since the TU sets the flow, the system operates at the max of the cooling and heating air flow rates.

Remember here there is a fan, a UnitarySystem w/ cooling coil, and UnitarySystem with heating coil all on the main branch.

    Component Sizing Information, AirTerminal:SingleDuct:Uncontrolled, ZONE1DIRECTAIR, Design Size Maximum Air Flow Rate [m3/s], 0.45099
    Component Sizing Information, AirTerminal:SingleDuct:Uncontrolled, ZONE2DIRECTAIR, Design Size Maximum Air Flow Rate [m3/s], 0.39470
    Component Sizing Information, AirTerminal:SingleDuct:Uncontrolled, ZONE3DIRECTAIR, Design Size Maximum Air Flow Rate [m3/s], 0.52813

    Component Sizing Information, AirLoopHVAC, HEAT PUMP SYS 1, Design Supply Air Flow Rate [m3/s], 1.37381 (RR - MAIN BRANCH FLOW)

    Component Sizing Information, Fan:SystemModel, SUPPLY FAN 1, Design Size Design Maximum Air Flow Rate [m3/s], 1.37381

    Component Sizing Information, AirLoopHVAC:UnitarySystem, DX COOLING COIL, Design Size Cooling Supply Air Flow Rate [m3/s], 1.37381
    Component Sizing Information, AirLoopHVAC:UnitarySystem, DX COOLING COIL, User-Specified No Load Supply Air Flow Rate [m3/s], 0.00000
    Component Sizing Information, Coil:Cooling:DX:SingleSpeed, HEAT PUMP ACDXCOIL 1, Design Size Rated Air Flow Rate [m3/s], 1.37381

    Component Sizing Information, AirLoopHVAC:UnitarySystem, HEATPUMP DX COIL 1, Design Size Heating Supply Air Flow Rate [m3/s], 0.47112
    Component Sizing Information, AirLoopHVAC:UnitarySystem, HEATPUMP DX COIL 1, User-Specified No Load Supply Air Flow Rate [m3/s], 0.00000
    Component Sizing Information, Coil:Heating:DX:SingleSpeed, HEAT PUMP DX HEATING COIL 1, Design Size Rated Air Flow Rate [m3/s], 0.47112

The air flow rate through each component on the main branch is the same, so that's a good thing. The bad thing is it's constant at the max flow rate in cooling and heating mode.

unitarysystem_dxcoilsystemauto

**Comment by rraustad** [comment]
The defect file appears to act as requested by the user.

6200-in_defect file

File: src/EnergyPlus/HVACUnitarySystem.cc
Comment (rraustad) [inline src/EnergyPlus/HVACUnitarySystem.cc:2082]: This line was overwriting the user input when a cooling coil did not exist negating the users attempt to set flow = 0.

File: src/EnergyPlus/HVACUnitarySystem.cc
Comment (rraustad) [inline src/EnergyPlus/HVACUnitarySystem.cc:2084]: This line was overwriting the user input when a heating coil did not exist negating the users attempt to set flow = 0.

File: doc/input-output-reference/src/overview/group-unitary-equipment.tex
Comment (mjwitte) [inline doc/input-output-reference/src/overview/group-unitary-equipment.tex]: Shouldn't this one talk about the cooling coil, not the heating coi?  Also, in the I/O ref descriptions for the fields after this, they all say "must be greater than 0" but zero is valid.  Same for heating and no-load flow fields.

File: doc/input-output-reference/src/overview/group-unitary-equipment.tex
Comment (mjwitte) [inline doc/input-output-reference/src/overview/group-unitary-equipment.tex]: Heating here, not cooling?

**Review by mjwitte — COMMENTED** [review]
Defect file is much better, but the furnace fan energy is showing 3.81E-09 instead of zero during cooling load hours. See 07/25 for example.

UnitarySystem_DXCoilSystemAuto is crashing in the [linux debug build](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-221206a01fc9d9bfffe50409b960951151bcc4c6-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html).  Probably a divide-by-zero.  Tried to reproduce it on Windows (with the fp:strict options) but it runs OK.  Will delete the linux results and re-run them.

File: doc/input-output-reference/src/overview/group-unitary-equipment.tex
Comment (rraustad) [reply doc/input-output-reference/src/overview/group-unitary-equipment.tex]: I was trying to talk about the other coil here. I guess that does sound confusing. I'll flip it to read like this in both places:

If the system does not have a cooling coil a 0 may be entered for cooling air flow rate and/or no load supply air flow rate to turn the fan off when cooling is not required.

**Comment by mjwitte** [comment]
@Myoldmopar Please run UnitarySystem_DXCoilSystemAuto in linux debug build to find out why it's [crashing](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-221206a01fc9d9bfffe50409b960951151bcc4c6-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html)

**Comment by Myoldmopar** [comment]
HVACUnitarySystem.cc:2194

**Comment by Myoldmopar** [comment]
SIGFPE - probably divide by zero on `UnitarySystem( UnitarySysNum ).MaxHeatAirVolFlow`

File: doc/input-output-reference/src/overview/group-unitary-equipment.tex
Comment (rraustad) [reply doc/input-output-reference/src/overview/group-unitary-equipment.tex]: "Must be greater than 0 if the xxxxing coil is present"

**Comment by mjwitte** [comment]
Hmm, so I should have been able to see that on Windows using the NAN debugging tips . . .  Thanks.  @rraustad can work on fixing it.

**Comment by rraustad** [comment]
Did either of you ad \type node to the ZoneHVAC:IdealLoadsAirSystem object for the new field System Inlet Air Node Name? I'll add it here so it's not forgotten.

**Comment by rraustad** [comment]
I'll wait for either of you to say something since the idd is locked down.

**Comment by mjwitte** [comment]
It's up to @Myoldmopar but I'd say not here - if we add it, it should probably be a separate pull request.

**Comment by rraustad** [comment]
defect file used this input:

    SupplyAirFlowRate,                      !- Cooling Supply Air Flow Rate Method
    0.000000000000001,                      !- Cooling Supply Air Flow Rate {m3/s}

**Comment by Myoldmopar** [comment]
Definitely don't put any changes to the IDD in this branch if you want it in for 8.8.  If it needs to be added, put it on a separate branch/PR, label it IDDChange, and mark it for the Spring milestone. :+1:

**Comment by rraustad** [comment]
Defect file is now able to run with 0 entered for cooling air flow rate (there were multiple <=0 checks) and fan energy is 0 now.

image

File: src/EnergyPlus/HVACUnitarySystem.cc
Comment (rraustad) [inline src/EnergyPlus/HVACUnitarySystem.cc:2198]: Linux failures were here with divide by 0.

File: src/EnergyPlus/HVACUnitarySystem.cc
Comment (rraustad) [inline src/EnergyPlus/HVACUnitarySystem.cc:5408]: And changes to GetInput to allow a 0 for air flow rate.

**Comment by rraustad** [comment]
I deleted the results for the Linux "Unhandled Failure" and that test passed this time around.

**Comment by mjwitte** [comment]
@rraustad This all looks good now.  @nmerket Would you like to review before merge?

**Review by nmerket — APPROVED** [review]
Looks good to me.

**Comment by nmerket** [comment]
Fixes #6200

**Comment by nmerket** [comment]
Thanks for doing this @rraustad!

**Pull request merged**
